contents/tutorials/nextjs-monitoring.md

```diff
@@ -0,0 +1,217 @@
+
+```js
+// app/provider.js
+'use client'
+import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
+
+if (typeof window !== 'undefined') {
+  posthog.init('<ph_project_api_key>', {
+    api_host: '<ph_instance_address>'
+  })
+}
+
+export default function PHProvider({ children }) {
+  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
+}
+```
+
+Once created, import and add the provider to your `layout.js` file. 
+
+```js
+// app/layout.js
+import './globals.css'
+import Providers from './providers'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <Providers>
+        <body>{children}</body>
+      </Providers>
+    </html>
+  )
+}
+```
+
+Once this is set up, go to your [project settings](https://app.posthog.com/project/settings), and make sure:
+
+- Autocapture is enabled
+- Under recordings, record user sessions, capture console logs, and capture network performance are all enabled.
+
+Enabling these provides the key monitoring tools including performance monitoring, app usage, and session replays. It also enables capturing custom events which we use for errors and more performance data next.
+
+## Custom error capture
+
+Another part of monitoring is error capture. There are a couple of ways to set this up with PostHog. First, you could set up and use our [Sentry integration](/docs/libraries/sentry). Second, you could set up custom error capture, which we show here.
+
+To do this, we can create a React error boundary. In the `app` folder, create a file named `error.js`. In this file, set up a basic component that uses `usePostHog` to capture an `$exception` event and returns an error page. It should look like this:
+
+```js
+'use client'
+import { usePostHog } from 'posthog-js/react'
+
+export default function Error({ error }) {
+  const posthog = usePostHog()
+  posthog?.capture('$exception', {
+    message: error.message,
+  })
+  
+  return (
+    <div>
+      <h1>There was an error</h1>
+    </div>
+  )
+}
+```
+
+Now, in your `app` folder, create a new folder named `about` and add a file named `page.js`. In this file, set up a component that errors like this:
+
+```js
+// app/about/page.js
+const session = null
+
+export default function About() {
+
+  if (!session) {
+    throw new Error('No session')
+  }
+
+  return (
+    <h1>Home</h1>
+  )
+}
+```
+
+When you go to `[http://localhost:3000/about](http://localhost:3000/about)`, you get the error page.
+
+![Error](../images/tutorials/nextjs-monitoring/app-error.png)
+
+This also captures an `exception` event in PostHog with details like the current URL, OS, browser, time, and more.
+
+![Exception](../images/tutorials/nextjs-monitoring/exception.png)
+
+This captures errors thrown from your app in PostHog, wherever they happen.
+
+## Custom performance capture
```

Comment:
Overall I enjoyed this section! Didn't know we could do this

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FQEYYlJqOaEhXrjTrOH%2Fgiphy.gif&token=nice" data-gitmeme-token="nice"><img src="https://media4.giphy.com/media/QEYYlJqOaEhXrjTrOH/giphy.gif" title="Created by gitme.me with /nice"/></a>